src/api/ledger/parse/settings.js

```diff
@@ -2,27 +2,69 @@
 'use strict';
 const _ = require('lodash');
 const assert = require('assert');
-const AccountSetFlags = require('./utils').constants.AccountSetFlags;
+const AccountFlags = require('./utils').constants.AccountFlags;
+const AccountFlagIndices = require('./utils').constants.AccountFlagIndices;
 const parseFields = require('./fields');
 
 function getName(flagNumber) {
-  return _.findKey(AccountSetFlags, (v) => v === flagNumber);
+  return _.findKey(AccountFlagIndices, (v) => v === flagNumber);
 }
 
-function parseSettings(tx: Object) {
-  const txType = tx.TransactionType;
-  assert(txType === 'AccountSet' || txType === 'SetRegularKey');
+function parseFlags(tx: Object) {
   const settings = {};
+  if (tx.TransactionType !== 'AccountSet') {
+    return settings;
+  }
+
+  const modifiedNodes = tx.meta.AffectedNodes.filter(node =>
+    node.ModifiedNode.LedgerEntryType === 'AccountRoot');
+  assert(modifiedNodes.length === 1);
+  const node = modifiedNodes[0].ModifiedNode;
+
+  const oldFlags = _.get(node.PreviousFields, 'Flags');
+  const newFlags = _.get(node.FinalFields, 'Flags');
+
+  if (oldFlags !== undefined && newFlags !== undefined) {
+    const xorFlags = oldFlags ^ newFlags;
+    const setFlags = newFlags & xorFlags;
+    const clearFlags = oldFlags & xorFlags;
+    _.forEach(AccountFlags, (flagValue, flagName) => {
+      if (setFlags & flagValue) {
+        settings[flagName] = true;
+      } else if (clearFlags & flagValue) {
+        settings[flagName] = false;
+      }
+    });
+  }
+
+  // enableTransactionIDTracking requires a special case because it
+  // does not affect the Flags field; instead it adds/removes a field called
+  // "AccountTxnID" to/from the account root. Checking SetFlag and ClearFlag
```

Comment:
I think you could just check for the AccountTxnID in the metadata for a more meaningful check of what the transaction actually changed. This would be in line with checking the Flags in the metadata as above rather than assuming the transaction did what was intended. Basically following the general rule of always looking at the metadata.